partiql-plan/src/main/kotlin/org/partiql/plan/v1/operator/rel/RelExcludeStep.kt

```diff
@@ -1,40 +1,103 @@
 package org.partiql.plan.v1.operator.rel
 
 /**
- * TODO DOCUMENTATION
+ * Logical EXCLUDE step, one of: index, key, symbol, struct wildcard, or collection wildcard.
  */
 public interface RelExcludeStep {
 
     public fun getSubsteps(): List<RelExcludeStep>
```

Comment:
It's a bit odd to use this modeling. Each `RelExcludeStep` contains a list of preceding steps, which each and every preceding step contains their own list of preceding steps. I'd potentially just make this as such:
```suggestion
    public fun getParent(): RelExcludeStep?
```

Alternatively, a `RelExcludeSteps` (or something similarly named) could hold the list of steps:
```kotlin
public interface RelExcludeSteps {
    public fun getSteps(): List<RelExcludeStep>
}

public interface RelExcludeStep
```

Reply:
I think the current modeling is fine.

> Each `RelExcludeStep` contains a list of preceding steps, which each and every preceding step contains their own list of preceding steps.

Not sure what you mean by preceding steps. The plan modeling is like a [prefix tree](https://en.wikipedia.org/wiki/Trie). A given `RelExcludeStep` stores a list of other exclude paths that share the same prefix (i.e. same exclude path till the given step).